frontend/src/features/projects/erp/forms/AgencyInterest.tsx

```diff
@@ -0,0 +1,123 @@
+import { AgencyResponses, FormikTable, IProject } from '../../common';
+import { getIn, useFormikContext } from 'formik';
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { RootState } from 'reducers/rootReducer';
+import { ILookupCode } from 'actions/lookupActions';
+import { ILookupCodeState } from 'reducers/lookupCodeReducer';
+import _ from 'lodash';
+import * as API from 'constants/API';
+import { ParentSelect } from 'components/common/form/ParentSelect';
+import { mapLookupCodeWithParentString } from 'utils';
+import { Button, Col, Row } from 'react-bootstrap';
+import { AgencyInterestColumns } from './AgencyInterestColumns';
+
+export interface IAgencyInterestProps {
+  /** Whether the controls are disabled. */
+  disabled?: boolean;
+}
+
+/**
+ * Returns a component to display a table containing agency interest responses.
+ * @param param0 Property attributes.
+ */
+export const AgencyInterest = ({ disabled = false }: IAgencyInterestProps) => {
+  const { values, setValues, setFieldValue } = useFormikContext<IProject>();
+  const [enableAdd, setEnableAdd] = React.useState(false);
+
+  const lookupCodes = useSelector<RootState, ILookupCode[]>(
+    state => (state.lookupCode as ILookupCodeState).lookupCodes,
+  );
+  const agencies = _.filter(lookupCodes, (lookupCode: ILookupCode) => {
+    return lookupCode.type === API.AGENCY_CODE_SET_NAME;
+  });
+  const agencyOptions = (agencies ?? []).map(c => mapLookupCodeWithParentString(c, agencies));
+
+  const onAddAgency = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
+    event.preventDefault();
+    const selectedAgency = getIn(values, 'addAgencyResponse');
+    const agency = agencies.find(a => a.id == selectedAgency?.value);
```

Comment:
`==` -> `===`

Reply:
Actually this was intentional, but looks like our linting won't allow it.  Essentially `a.id` in typescript is a string, but during runtime it is a number.  Which means `===` will never be `true`...  I'll modify to handle this and make the code way harder to read...